dev-libs/libffi/libffi-3.4.2-r1.ebuild

```diff
@@ -10,7 +10,8 @@ MY_P=${PN}-${MY_PV}
 
 DESCRIPTION="a portable, high level programming interface to various calling conventions"
 HOMEPAGE="https://sourceware.org/libffi/"
-SRC_URI="https://github.com/libffi/libffi/releases/download/v${MY_PV}/${MY_P}.tar.gz"
+SRC_URI="https://github.com/libffi/libffi/releases/download/v${MY_PV}/${MY_P}.tar.gz
+	https://dev.gentoo.org/~xen0n/distfiles/${MY_P}-loongarch64-20220425.patch.xz"
```

Comment:
This touches so much stuff outside loongarch, you should probably gate it with USE=experimental-loong. 
In addition, the patch touches automake source files, so you need to run eautoreconf from autotools.eclass in src_prepare.

Reply:
1. yes
2. libffi is early enough that we _might_ have to avoid it (eautoreconf) given python needs it

Reply:
Sure enough; libffi can't autoconf due to `autoconf -> m4 -> gettext -> libxml2[python] -> python -> libffi`. I'll adjust this.

Reply:
Done. (gating with `USE=experimental-loong` and verifying that autoconf isn't invoked during build)